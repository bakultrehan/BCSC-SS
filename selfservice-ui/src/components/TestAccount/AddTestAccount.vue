/** * Add TestAccount component */

<template>
  <v-card class="mx-auto " flat="">
    <v-item-group mandatory>
      <v-container>
        <v-row class="mx-4">
          <v-col cols="12" flat>
            <v-card flat>
              <div
                class="text-left"
                v-html="$t('AddTestAccount.pageinfo')"
              ></div>
            </v-card>
          </v-col>

          <v-col class="col-12" v-if="errorStatus || successStatus">
            <Alert
              type="error"
              v-if="errorStatus"
              class="alert-top text-left"
              >{{ $t('AddTestAccount.errorMessage') }}</Alert
            >
            <Alert
              type="success"
              class="alert-top text-left"
              v-if="successStatus"
            >
              {{ successCount.created }}
              {{ $t('AddTestAccount.successCreatedMessage') }}
              <br />
              {{ successCount.skipped }}
              {{ $t('AddTestAccount.successSkippedMessage') }}
            </Alert>
          </v-col>
          <v-col cols="12" flat>
            <v-card flat>
              <TextArea
                v-model="testAccounts"
                :label="$t('AddTestAccount.special_notes')"
                type="text"
                outlined
                rows="10"
                name="test-account-text"
                id="test-account-text"
                data-test-id="test-account-text"
              />
            </v-card>
          </v-col>
        </v-row>
      </v-container>
    </v-item-group>
    <v-col cols="12">
      <v-card flat>
        <v-divider></v-divider>
        <v-card-actions class="mx-4">
          <v-spacer></v-spacer>

          <Button
            :disabled="testAccounts === ''"
            :loading="isLoading"
            class="white--text submit-test-account ml-6"
            depressed
            @click="submitTestAccount"
            name="btn-test-account"
            data-test-id="btn-test-account"
            >{{ $t('AddTestAccount.btnSaveChanges') }}</Button
          >
        </v-card-actions>
      </v-card>
    </v-col>
  </v-card>
</template>
<script lang="ts">
import { Component, Vue, Watch, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Button from '@/Atomic/Button/Button.vue';
import TextArea from '@/Atomic/TextArea/TextArea.vue';
import Alert from '@/Atomic/Alert/Alert.vue';

const TestAccountModule = namespace('TestAccountModule');

@Component({
  components: {
    Button,
    TextArea,
    Alert,
  },
})
export default class TestAccount extends Vue {
  @Prop({ default: 0 })
  public id!: number;

  @TestAccountModule.Action('addTestAccounts')
  public addTestAccounts!: any;
  @TestAccountModule.Getter('isLoading')
  public isLoading!: boolean;

  @TestAccountModule.Action('clearStatus') public clearStatus!: any;
  @TestAccountModule.Getter('getTestAccountSuccessData')
  public getTestAccountSuccessData!: any;

  private testAccounts: string = '';
  private successCount: any = { created: 0, skipped: 0 };
  private successStatus: boolean = false;
  private errorStatus: boolean = false;

  @Watch('getTestAccountSuccessData')
  private ongetTestAccountSuccessDataChanged(val: any) {
    const { successStatus, errorStatus, testAccountSuccessData } = val;
    this.clearAllStatus();
    if (successStatus) {
      this.successStatus = true;
      this.successCount = testAccountSuccessData;
    } else if (errorStatus) {
      this.errorStatus = true;
    }
    setTimeout(() => {
      this.clearAllStatus();
      this.clearStatus();
    }, 10000);
  }

  private submitTestAccount() {
    this.addTestAccounts({
      testAccounts: this.testAccounts,
    });
  }
  private clearAllStatus() {
    this.errorStatus = false;
    this.successStatus = false;
  }

  private mounted() {
    this.clearAllStatus();
    this.clearStatus();
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
</style>
