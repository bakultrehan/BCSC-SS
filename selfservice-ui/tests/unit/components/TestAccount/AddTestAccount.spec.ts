import { shallowMount, mount } from '@vue/test-utils';
import AddTestAccount from '@/components/TestAccount/AddTestAccount.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
let vuetify: any;
vuetify = new Vuetify();

describe('AddTestAccount.vue', () => {
  const store = new Vuex.Store({
    modules: {
      AddTestAccountModule: {
        namespaced: true,
        state: {},
        getters: {
          successStatus: jest.fn(),
          errorStatus: jest.fn(),
          isLoading: jest.fn(),
        },
        actions: {
          addAddTestAccounts: jest.fn(),
          clearStatus: jest.fn(),
        },
      },
      TestAccountModule: {
        namespaced: true,
        state: {},
        getters: {
          getTestAccountCount: jest.fn(),
          isLoading: jest.fn(),
          getTestAccountSuccessData: jest.fn(),
        },
        actions: {
          loadTestAccountCount: jest.fn(),
          clearStatus: jest.fn(),
          addTestAccounts: jest.fn(),
        },
      },
    },
  });

  const mountFunction = (options: any) => {
    return mount(AddTestAccount, {
      store,
      vuetify,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options,
    });
  };

  it('renders props when passed with gettors', () => {
    const AddTestAccountPage = mountFunction({});
    expect(AddTestAccountPage.element).toMatchSnapshot();
  });

  it('renders props when delete member', () => {
    const AddTestAccountPage = mountFunction({});
    const toggleAddMember = jest.fn();

    AddTestAccountPage.setData({ testAccounts: 'test' });
    const button = AddTestAccountPage.find('.submit-test-account');
    AddTestAccountPage.vm.$on('action-btn:clicked', toggleAddMember);
    button.trigger('click');

    expect(AddTestAccountPage.element).toMatchSnapshot();
  });
});
