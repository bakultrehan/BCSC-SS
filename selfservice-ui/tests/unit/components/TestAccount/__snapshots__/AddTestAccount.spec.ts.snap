// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTestAccount.vue renders props when passed with gettors 1`] = `
<v-card-stub
  class="mx-auto "
  flat="true"
  loaderheight="4"
  tag="div"
>
  <v-item-group-stub
    activeclass="v-item--active"
    mandatory="true"
  >
    <v-container-stub
      tag="div"
    >
      <v-row-stub
        class="mx-4"
        tag="div"
      >
        <v-col-stub
          cols="12"
          flat=""
          tag="div"
        >
          <v-card-stub
            flat="true"
            loaderheight="4"
            tag="div"
          >
            <div
              class="text-left"
            />
          </v-card-stub>
        </v-col-stub>
         
        <!---->
         
        <v-col-stub
          cols="12"
          flat=""
          tag="div"
        >
          <v-card-stub
            flat="true"
            loaderheight="4"
            tag="div"
          >
            <textarea-stub
              counter=""
              data-test-id="test-account-text"
              helptext=""
              id="test-account-text"
              label=""
              name="test-account-text"
              outlined=""
              rows="10"
              type="text"
              value=""
            />
          </v-card-stub>
        </v-col-stub>
      </v-row-stub>
    </v-container-stub>
  </v-item-group-stub>
   
  <v-col-stub
    cols="12"
    tag="div"
  >
    <v-card-stub
      flat="true"
      loaderheight="4"
      tag="div"
    >
      <v-divider-stub />
       
      <v-card-actions-stub
        class="mx-4"
      >
        <v-spacer-stub />
         
        <button-stub
          class="white--text submit-account ml-6"
          color=""
          data-test-id="btn-test-account"
          depressed=""
          disabled="true"
          name="btn-test-account"
          type="button"
        >
          
        </button-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-col-stub>
</v-card-stub>
`;
